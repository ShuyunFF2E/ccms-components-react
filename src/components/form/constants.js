import { Children } from 'react';

export const LAYOUT_TYPES = {
	HORIZONTAL: 'horizontal',
	VERTICAL: 'vertical',
	INLINE: 'inline'
};

export const LABEL_ALIGN = {
	LEFT: 'left',
	RIGHT: 'right'
};

export const DATA_FIELD = 'data-field';

export const findFieldsName = (children = []) => {
	const result = [];

	Children.toArray(children).forEach(child => {
		const { props } = child;

		if (props) {
			if (props[DATA_FIELD]) {
				result.push(props[DATA_FIELD]);
			} else if (props.children && Children.count(props.children || [])) {
				result.push(...findFieldsName(props.children));
			}
		}
	});

	return result;
};

export const getNamesByNode = parentNode => {
	const children = parentNode.querySelectorAll(`[${DATA_FIELD}]`);

	return [...children].map(child => child.getAttribute(DATA_FIELD));
};

export const findDestroyedFields = (prevNames, names) => {
	return prevNames.reduce((acc, n) => {
		if (!names.includes(n)) acc.push(n);

		return acc;
	}, []);
};
