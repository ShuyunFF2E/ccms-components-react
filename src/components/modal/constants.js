export const sandboxSelector = 'r-w_s-';

export default {};
