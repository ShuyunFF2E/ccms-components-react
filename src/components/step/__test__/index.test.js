import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import { render, mount } from 'enzyme';
import { prefixCls } from '@utils';
import mountTest from '@tests/shared/mountTest';
import Step from '../index';
import { HORIZONTAL, VERTICAL, INLINE, CIRCLE, DOT } from '../constants';

const classSelector = `${prefixCls}-step`;
const LAYOUTS = [HORIZONTAL, VERTICAL, INLINE];
const TYPES = [CIRCLE, DOT];

class StepApp extends Component {
	render() {
		return (
			<Step {...this.props}>
				<Step.Item title="把冰箱门打开" />
				<Step.Item title="把大象放进去" />
			</Step>
		);
	}
}

describe('Step', () => {
	mountTest(Step);

	it('renders correctly', () => {
		const wrapper = render(<StepApp current={0} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('mount correctly', () => {
		expect(() => renderer.create(<StepApp current={0} />)).not.toThrow();
	});

	it('should active current step when current set', () => {
		const wrapper = mount(<StepApp current={1} />);
		expect(
			wrapper
				.find(`.${classSelector}-icon`)
				.at(1)
				.hasClass('process')
		).toBeTruthy();
	});

	it('renders different direction correctly', () => {
		LAYOUTS.forEach(layout => {
			const wrapper = render(<StepApp direction={layout} />);
			expect(wrapper).toMatchSnapshot();
		});
	});

	it('renders different type correctly', () => {
		TYPES.forEach(type => {
			const wrapper = render(<StepApp type={type} />);
			expect(wrapper).toMatchSnapshot();
		});
	});

	it('className work correctly', () => {
		const wrapper = mount(<StepApp className="step-test" />);
		expect(
			wrapper
				.find(`.${classSelector}`)
				.at(0)
				.hasClass('step-test')
		).toBeTruthy();
	});

	it('onClick work correctly', () => {
		const onClick = jest.fn();
		const wrapper = mount(<StepApp onClick={onClick} />);
		wrapper
			.find(`.${classSelector}-icon`)
			.at(1)
			.simulate('click');
		expect(onClick).toHaveBeenCalledWith(1);
	});
});
