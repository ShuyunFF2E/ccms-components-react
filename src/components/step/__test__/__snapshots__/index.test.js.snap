// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step renders correctly 1`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;

exports[`Step renders different direction correctly 1`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;

exports[`Step renders different direction correctly 2`] = `
<div
  class="cloud-step vertical circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;

exports[`Step renders different direction correctly 3`] = `
<div
  class="cloud-step inline circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;

exports[`Step renders different type correctly 1`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;

exports[`Step renders different type correctly 2`] = `
<div
  class="cloud-step horizontal dot"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        2
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把大象放进去
      </div>
    </div>
  </div>
</div>
`;
